socialhome/streams/views.py

```diff
@@ -10,12 +12,67 @@ class BaseStreamView(ListView):
     ordering = "-created"
     paginate_by = 30
     stream_name = ""
+    vue = False
+    queryset = []
+
+    def dispatch(self, request, *args, **kwargs):
+        self.vue = bool(request.GET.get("vue", False))
+        return super().dispatch(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["stream_name"] = self.stream_name
+        if self.vue:
+            context["json_context"] = self._define_context_json(self.stream_name)
         return context
 
+    def _define_context_json(self, stream_name):
+        request_user_profile = getattr(self.request.user, "profile", None)
+
+        def dump_content(content):
+            is_user_author = (request_user_profile is not None and content.author == request_user_profile)
```

Comment:
Just `content.author == request_user_profile` should be enough, because if the latter is `None` - they're not equal.